app/src/main/java/app/revanced/manager/ui/component/LoadingIndicator.kt

```diff
@@ -1,37 +1,39 @@
 package app.revanced.manager.ui.component
 
-import androidx.compose.foundation.layout.Arrangement
-import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.foundation.layout.padding
 import androidx.compose.material3.CircularProgressIndicator
-import androidx.compose.material3.Text
+import androidx.compose.material3.ProgressIndicatorDefaults
 import androidx.compose.runtime.Composable
-import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.unit.dp
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.StrokeCap
+import androidx.compose.ui.unit.Dp
 
 @Composable
 fun LoadingIndicator(
     modifier: Modifier = Modifier,
-    progress: Float? = null,
-    text: String? = null
+    progress: (() -> Float?)? = null,
```

Comment:
Why was this switched to a function?

Reply:
Then why is `progress` being called once and then turned into a lambda again later in `LoadingIndicator` instead of passing it directly?

Reply:
Because both the function and the output of the function can be null, i did this so it would be more readable when its used.

Reply:
No because `progress` doesn't accept null

Reply:
Wouldn't it be possible to make `progress` non-nullable with a default that returns null then?

Reply:
`CircularProgressionIndicator` doesnt accept null, thats why im using 2 separate composables for when the progress is null or not

Reply:
Alright but then why are both the return value **and** lambda nullable?
Feels like a non-nullable lambda that returns a nullable float would work